### PR TITLE
Add Integration Test for GCF Adapter

### LINKED ISSUE #7: Create tests that run function invoker
Investigate if it's possible to run function:run from tests, and create tests that run.

### PR DESCRIPTION
**Pull request by dzou**
This adds an integration test for the GCF adapter equivalent to running `mvn function:run` through the plugin.

Fixes #7.

### CODE REVIEW

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml [hunk 1]
```diff
@@ -45,6 +45,18 @@
...
 
-	</dependencies>
+		<dependency>
+			<groupId>com.google.cloud.functions.invoker</groupId>
+			<artifactId>java-function-invoker</artifactId>
+			<version>1.0.0-alpha-2-rc3</version>
+			<scope>test</scope>
+		</dependency>
 
+		<dependency>
```
Comment (dmitry-s) [inline spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml]: we already have this dependency

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml [hunk 2]
```diff
@@ -45,6 +45,18 @@
 			<scope>test</scope>
 		</dependency>
 
-	</dependencies>
+		<dependency>
+			<groupId>com.google.cloud.functions.invoker</groupId>
+			<artifactId>java-function-invoker</artifactId>
+			<version>1.0.0-alpha-2-rc3</version>
```
Comment (dmitry-s) [inline spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml]: make it a property or use `google.cloud.functions.version` one

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/GcfIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	private static final String ADAPTER_CLASS = GcfSpringBootHttpRequestHandler.class.getCanonicalName();
+
+	@BeforeClass
+	public static void setup() {
+		System.setProperty("MAIN_CLASS", MAIN_CLASS);
+		runCloudFunctionServer();
+	}
+
+	@Test
+	public void test() {
```
Comment (dmitry-s) [inline spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/GcfIntegrationTest.java]: put tests to a separate class

**Review by dmitry-s — DISMISSED** [review]
overall looks good, just a couple of small issues

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml (hunk 1 above)
Comment (dzou) [reply spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml]: done.

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml (hunk 2 above)
Comment (dzou) [reply spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml]: done.

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/GcfIntegrationTest.java (hunk 3 above)
Comment (dzou) [reply spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/GcfIntegrationTest.java]: done.

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/HttpFunctionIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+	// The Cloud Functions adapter class to serve
+	private static final Class<?> ADAPTER_CLASS = GcfSpringBootHttpRequestHandler.class;
+
+	// The main class which define the Spring Cloud Functions
+	private static final Class<?> MAIN_CLASS = CloudFunctionMain.class;
+
+	@BeforeClass
+	public static void setup() {
+		CloudFunctionServerUtils.startServer(PORT, ADAPTER_CLASS, MAIN_CLASS);
```
Comment (dmitry-s) [inline spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/HttpFunctionIntegrationTest.java]: Maybe move it to the parent class?
Also, do we need to shut down the service?

Comment (dzou) [reply spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/HttpFunctionIntegrationTest.java]: I ended up writing a ClassRule for the test which starts up and stops the server; I initially tried the inheritance route for the test but felt that the classrule looked cleaner.

And done - the rule now shuts down the service.

**Review by meltsufin — APPROVED** [review]

**Pull request merged**